### PR TITLE
API-1287: Technical stack to send Webhooks

### PR DESCRIPTION
**Pull request by pchasle**
Technical stack to send webhooks includes:
- [x] Create a Command Handler to send a Business Event through Webhook
- [x] Retrieve all the connections' webhooks
    - Webhook URL must exist and Webhook should be enabled + integration tests
- [x] Call Webhooks via Guzzle Pool
    - Should supports sending 50+ payloads of 10+ mb
- [x] Log every call success or failure (uuid + datetime + success/failure)

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilderRegistry.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Application\Webhook;
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilderRegistry.php:4]: Move to `Akeneo\Tool\Component\Api\Webhook`

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Application\Webhook;
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php:4]: Move to `Akeneo\Tool\Component\Api\Webhook`

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /** @var WebhookEventBuilderRegistry */
+    private $dataBuilderRegistry;
+
+    public function __construct(WebhookEventBuilderRegistry $dataBuilderRegistry)
+    {
+        $this->dataBuilderRegistry = $dataBuilderRegistry;
+    }
+
+    public function build(ConnectionWebhook $webhook, BusinessEventInterface $businessEvent): WebhookEvent
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/WebhookEventBuilder.php:20]: Create a `WebhookConfigInterface`
```suggestion
    public function build(ConnectionWebhook $webhook, object $businessEvent): WebhookEvent
```

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectConnectionsWebhookIntegration.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+use Akeneo\Connectivity\Connection\Domain\Settings\Model\Read\ConnectionWithCredentials;
+use Akeneo\Connectivity\Connection\Domain\Settings\Model\ValueObject\FlowType;
+use Akeneo\Connectivity\Connection\Domain\Webhook\Persistence\Query\SelectConnectionsWebhookQuery;
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Integration\TestCase;
+use Doctrine\DBAL\Connection as DbalConnection;
+use PHPUnit\Framework\Assert;
+
+/**
+ * @package   Akeneo\Connectivity\Connection\back\tests\Integration\Persistence\Dbal\Query
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectConnectionsWebhookIntegration.php:18]: Remove @package from your code pls

File: src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectConnectionsWebhookIntegration.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->connectionLoader = $this->get('akeneo_connectivity.connection.fixtures.connection_loader');
+        $this->selectConnectionsWebhookQuery = $this->get('akeneo_connectivity_connection_persistence_query_select_connections_webhook');
+        $this->dbalConnection = $this->get('database_connection');
+    }
+
+    public function test_it_finds_a_connection_with_enabled_webhooks()
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Integration/Persistence/Dbal/Query/DbalSelectConnectionsWebhookIntegration.php:43]: you must test that it returns an empty array if there is no result.

File: src/Akeneo/Pim/Enrichment/Component/Product/WebhookEventBuilder/ProductEventDataBuilder.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+use Akeneo\Pim\Enrichment\Bundle\Message\ProductCreated;
+use Akeneo\Pim\Enrichment\Bundle\Message\ProductUpdated;
+use Akeneo\Platform\Component\EventQueue\BusinessEventInterface;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductEventDataBuilder implements WebhookEventDataBuilder
```
Comment (willy-ahva) [inline src/Akeneo/Pim/Enrichment/Component/Product/WebhookEventBuilder/ProductEventDataBuilder.php:17]: spec

File: src/Akeneo/Tool/Bundle/WebhookBundle/spec/Client/RequestSenderSpec.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                    'PACK' => ['groups' => [], 'products' => ['bar', 'baz'], 'product_models' => []],
+                    'UPSELL' => ['groups' => ['groupA'], 'products' => [], 'product_models' => []],
+                    'X_SELL' => ['groups' => ['groupB'], 'products' => ['bar'], 'product_models' => []],
+                    'SUBSTITUTION' => ['groups' => [], 'products' => [], 'product_models' => []],
+                ],
+                'quantified_associations' => [],
+            ],
+        ];
+    }
+}
```
Comment (willy-ahva) [inline src/Akeneo/Tool/Bundle/WebhookBundle/spec/Client/RequestSenderSpec.php:132]: blank line

File: src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Persistence/Query/SelectConnectionsWebhookQuery.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Connectivity\Connection\Domain\Webhook\Persistence\Query;
+
+interface SelectConnectionsWebhookQuery
+{
+    /**
+     * @return array<array{code: string, webhook: string}>
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Domain/Webhook/Persistence/Query/SelectConnectionsWebhookQuery.php:10]: it returns an array of ConnectionWebhook

**Review by tomglvng — APPROVED** [review]

**Pull request merged**